### PR TITLE
functions `take/drop/head -map-commute` added to  stdl Data.List.Properties

### PR DESCRIPTION
**Pull request by Sofia-Insa**
New functions added to stdl `Data.List.Properties `
   `take-map-commute`
   `drop-map-commute`
and
   `head-map-commute` ,
 `head-map-commute` was put in a new section 'head' of `Data.List.Properties`
 the import of `Data.Maybe` has been modified so `map` from `Data.List` isn't confused with `map` from `Data.Maybe`

changes declared in CHANGELOG.md

### CODE REVIEW

**Comment by MatthewDaggitt** [comment]
Please see comments in https://github.com/agda/agda-stdlib/pull/1983 for similar things that need addressing.

**Comment by jamesmckinna** [comment]
See also #509 and related issues #501 about `-commute` style naming: the library generally deprecates explicit use of `f-g-commute` in favour of simply `f-g` as the name of such lemmas.

**Comment by JacquesCarette** [comment]
Oops, I'd forgotten about the dropping of `-commute` and named functions like that in my private library (where these functions are being ported from). I'll have to back-port!

File: src/Data/List/Properties.agda [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (MatthewDaggitt) [inline src/Data/List/Properties.agda:1]: Can we remove the empty line here?

File: src/Data/List/Properties.agda [hunk 2]
```diff
@@ -758,6 +760,13 @@ length-take zero    xs       = refl
 length-take (suc n) []       = refl
 length-take (suc n) (x ∷ xs) = cong suc (length-take n xs)
 
+-- Take commutes with map.
+take-map : {f : A → B} → (n : ℕ) (xs : List A) → take n (map f xs) ≡ map f (take n xs)
```
Comment (MatthewDaggitt) [inline src/Data/List/Properties.agda:764]: Also no need for the first `→`

File: src/Data/List/Properties.agda [hunk 3]
```diff
@@ -766,6 +775,13 @@ length-drop zero    xs       = refl
 length-drop (suc n) []       = refl
 length-drop (suc n) (x ∷ xs) = length-drop n xs
 
+-- Drop commutes with map.
+drop-map : {f : A → B} → (n : ℕ) (xs : List A) → drop n (map f xs) ≡ map f (drop n xs)
```
Comment (MatthewDaggitt) [inline src/Data/List/Properties.agda:779]: Likewise here.

File: src/Data/List/Properties.agda [hunk 4]
```diff
@@ -766,6 +775,13 @@ length-drop zero    xs       = refl
...
+drop-map zero xs = refl
+drop-map (suc n) [] = refl
+drop-map (suc n) (a ∷ xs) = drop-map n xs
+
+
```
Comment (MatthewDaggitt) [inline src/Data/List/Properties.agda:784]: No need for double empty line here.

**Review by MatthewDaggitt — COMMENTED** [review]
Thanks for the changes. A couple more comments!

**Pull request merged**